inc/datahandlers/post.php

```diff
@@ -618,8 +628,6 @@ function verify_prefix()
 		}
 		else
 		{
-			$prefix_cache = build_prefixes($prefix);
```

Comment:
> Also, if I'm not mistaken, the negation here is wrong: https://github.com/mybb/mybb/blob/feature/inc/datahandlers/post.php#L696 since $num_prefixes should be set true only when a prefix can be used by the user.

That's a seperate issue which I won't fix here.

> And $user may be already set above: https://github.com/mybb/mybb/blob/feature/inc/datahandlers/post.php#L667-675 so an empty() check in the linked line(s) will possibly save a query.

`get_user` caches users so there it won't run a second query.